### PR TITLE
Remove some redundant casts

### PR DESCRIPTION
**Pull request by Gymnasiast**

### CODE REVIEW

File: src/openrct2/ride/Vehicle.cpp [hunk 1]
```diff
@@ -5462,8 +5462,7 @@ static void vehicle_update_crash(rct_vehicle * vehicle)
 
         invalidate_sprite_2((rct_sprite *)curVehicle);
 
-        LocationXYZ16 curPosition = { static_cast<sint16>(curVehicle->x), static_cast<sint16>(curVehicle->y),
-                                      static_cast<sint16>(curVehicle->z) };
+        LocationXYZ16 curPosition = {(curVehicle->x), (curVehicle->y), (curVehicle->z)};
```
Comment (janisozaur) [inline src/openrct2/ride/Vehicle.cpp]: remove braces

Comment (IntelOrca) [reply src/openrct2/ride/Vehicle.cpp]: brackets*

File: src/openrct2/world/Duck.cpp [hunk 2]
```diff
@@ -88,7 +88,7 @@ rct_duck * rct_sprite::AsDuck()
     rct_duck * result = nullptr;
     if (IsDuck())
     {
-        return (rct_duck *)result;
+        return result;
```
Comment (duncanspumpkin) [inline src/openrct2/world/Duck.cpp]: How does this even work?

Comment (IntelOrca) [reply src/openrct2/world/Duck.cpp]: should be `this`

File: src/openrct2/ride/Vehicle.cpp (hunk 1 above)
Comment (janisozaur) [reply src/openrct2/ride/Vehicle.cpp]: that thing

File: src/openrct2/world/Duck.cpp (hunk 2 above)
Comment (janisozaur) [reply src/openrct2/world/Duck.cpp]: Oh my

File: src/openrct2/ride/Vehicle.cpp (hunk 1 above)
Comment (AaronVanGeffen) [reply src/openrct2/ride/Vehicle.cpp]: parentheses ;)

File: src/openrct2/world/Duck.cpp (hunk 2 above)
Comment (IntelOrca) [reply src/openrct2/world/Duck.cpp]: Why did no one notice ducks weren't working properly?

Comment (ethanhs) [reply src/openrct2/world/Duck.cpp]: `return (🦆 *) this`

Comment (janisozaur) [reply src/openrct2/world/Duck.cpp]: This will need network version change

Comment (AaronVanGeffen) [reply src/openrct2/world/Duck.cpp]: @IntelOrca It doesn't look like this function is called anywhere yet.

File: src/openrct2/ride/Vehicle.cpp (hunk 1 above)
Comment (Gymnasiast) [reply src/openrct2/ride/Vehicle.cpp]: Removed the curved little lines ;-)

File: src/openrct2/world/Duck.cpp (hunk 2 above)
Comment (Gymnasiast) [reply src/openrct2/world/Duck.cpp]: Well, I'll be ducked.

File: src/openrct2/world/Duck.cpp [hunk 3]
```diff
@@ -88,7 +88,7 @@ rct_duck * rct_sprite::AsDuck()
     rct_duck * result = nullptr;
     if (IsDuck())
     {
-        return result;
+        return (rct_duck *)this;
```
Comment (janisozaur) [inline src/openrct2/world/Duck.cpp]: This will require network version change

Comment (Gymnasiast) [reply src/openrct2/world/Duck.cpp]: No, it doesn't. This function is not used.

Comment (janisozaur) [reply src/openrct2/world/Duck.cpp]: ok

**Pull request merged**